bluebutton-data-pipeline-rif-extract/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/extract/s3/task/ManifestEntryDownloadTask.java

```diff
@@ -63,17 +66,27 @@ public ManifestEntryDownloadResult call() throws Exception {
 			GetObjectRequest objectRequest = new GetObjectRequest(options.getS3BucketName(),
 					String.format("%s/%s/%s", DataSetMonitorWorker.S3_PREFIX_PENDING_DATA_SETS,
 							manifestEntry.getParentManifest().getTimestampText(), manifestEntry.getName()));
+
 			Path localTempFile = Files.createTempFile("data-pipeline-s3-temp", ".rif");
 
 			Timer.Context downloadTimer = appMetrics
 					.timer(MetricRegistry.name(getClass().getSimpleName(), "downloadSystemTime")).time();
-			LOGGER.debug("Downloading '{}' to '{}'...", manifestEntry, localTempFile.toAbsolutePath().toString());
+			LOGGER.info("Downloading '{}' to '{}'...", manifestEntry, localTempFile.toAbsolutePath().toString());
 			Download downloadHandle = s3TaskManager.getS3TransferManager().download(objectRequest,
 					localTempFile.toFile());
 			downloadHandle.waitForCompletion();
-			LOGGER.debug("Downloaded '{}' to '{}'.", manifestEntry, localTempFile.toAbsolutePath().toString());
+			LOGGER.info("Downloaded '{}' to '{}'.", manifestEntry, localTempFile.toAbsolutePath().toString());
 			downloadTimer.close();
 
+			// generate MD5ChkSum value on file just downloaded
+			String generatedMD5ChkSum = getMD5ChkSum(localTempFile);
+
+			String downloadedFileMD5ChkSum = downloadHandle.getObjectMetadata().getUserMetaDataOf("md5chksum");
+
+			if (!generatedMD5ChkSum.equals(downloadedFileMD5ChkSum))
```

Comment:
We have a lot of existing sample data that doesn't have checksums. I'd do two things:
1. File a new JIRA story to add checksums to all existing test and sample data. Will include adding code to `DataSetSubsetter` to do the S3 uploads itself.
2. Add a guard here to not blow up if a checksum isn't present at all, with a `TODO` referencing the above JIRA issue to remove the guard once it's no longer needed.